mne/annotations.py

```diff
@@ -701,9 +701,47 @@ def _ensure_annotation_object(obj):
                          'mne.Annotations. Got %s.' % obj)
 
 
+def _select_annotations_based_on_description(descriptions, event_id=None,
+                                             regexp='.*'):
+    """Get a collection of descriptions and returns index of selected."""
+    regexp_comp = re.compile(regexp)
+
+    if event_id is None:
+        event_id = Counter()
+
+    event_id_ = dict()
+    dropped = []
+    for desc in descriptions:
+        if desc in event_id_:
+            continue
+
+        if regexp_comp.match(desc) is None:
+            continue
+
+        if isinstance(event_id, dict):
+            if desc in event_id:
+                event_id_[desc] = event_id[desc]
+            else:
+                continue
+        else:
+            trigger = event_id(desc)
+            if trigger is not None:
+                event_id_[desc] = trigger
+            else:
+                dropped.append(desc)
+
+    event_sel = [ii for ii, kk in enumerate(descriptions)
+                 if kk in event_id_]
+
+    if len(event_sel) == 0 and regexp is not None:
```

Comment:
actually needs to compare with the default regex `'.*'`, and it's not even covered. I'll add this `on_missing` thing. Do you know how we test the logging from the top of your head?

Reply:
I wouldn't bother testing if the `logger.info` is actually emitted, it's probably overkill. If you test that by default it errors, and that with `warning` it warns, and that with `info` (or whatever we call it) it does neither (which can be done just by a smoke test since an un-caught error or warning would be a test error) it should be fine.

But if you really want to go for it you can use `mne.utils.catch_logging`

Reply:
since they are both the same, now we could factor them and have it as a `good first issue` make it into a function and cover it fully so that we don't have to go in the `'error'` etc.